erpnext/stock/get_item_details.py

```diff
@@ -320,6 +318,39 @@ def get_price_list_rate(args, item_doc, out):
 			out.update(get_last_purchase_details(item_doc.name,
 				args.name, args.conversion_rate))
 
+def get_item_price_name(args):
+	conditions = " WHERE item_code = '%s' AND price_list = '%s' AND min_qty <= '%s' AND uom = '%s'" \
+				 % frappe.db.escape(args.item_code, args.price_list, args.qty, args.uom)
+	if args.customer and not args.supplier:
+		conditions += "AND customer= '%s'" % frappe.db.escape(args.customer)
+
+	if args.supplier and not args.customer:
+		conditions += "AND supplier= '%s'" % frappe.db.escape(args.supplier)
+
+	return frappe.db.sql("""
+		SELECT name
+		FROM `tabItem Price`
+			%s """  % (conditions))
+
+def get_item_price(args, item_code):
+	conditions = "WHERE item_code = '%s' " \
+				 "AND price_list = '%s' " \
+				 "AND min_qty <= '%s' " \
+				 "AND uom = '%s' " \
+				 "AND (valid_from is null or valid_from <= '%s') " \
+				 "AND (valid_upto is null or valid_upto >= '%s')" \
+				  % (item_code, args.price_list, args.qty, args.uom, args.transaction_date, args.transaction_date)
+
+	if args.customer and not args.supplier:
+		conditions += "AND customer= '%s'" % frappe.db.escape(args.customer, percent=False)
```

Comment:
should have an OR condition here, so that if customer is not set in `Item Price` it should also select